titus-server-master/src/main/java/com/netflix/titus/master/clusteroperations/ClusterAgentAutoScaler.java

```diff
@@ -108,12 +109,16 @@
     private static final Comparator<AgentInstanceGroup> PREFER_ACTIVE_INSTANCE_GROUP_COMPARATOR = Comparator.comparing(ig -> ig.getLifecycleStatus().getState());
     private static final Comparator<AgentInstanceGroup> PREFER_PHASED_OUT_INSTANCE_GROUP_COMPARATOR = PREFER_ACTIVE_INSTANCE_GROUP_COMPARATOR.reversed();
     private static final Set<String> IGNORED_HARD_CONSTRAINT_NAMES = asSet("machineid", "machinetype");
+    private static final Set<FailureKind> IGNORED_FAILURE_KINDS_WITH_LAUNCHGUARD = ImmutableSet.<FailureKind>builder()
+            .addAll(FailureKind.NEVER_TRIGGER_AUTOSCALING)
+            .add(FailureKind.LaunchGuard)
```

Comment:
Can `LaunchGuard` be moved to `NEVER_TRIGGER_AUTOSCALING`?

Reply:
No, because we still autoscale in some cases even when all failures are of the `LaunchGuard` kind, e.g.: when breaching task SLOs